### PR TITLE
Mvj 454 hds login

### PR DESCRIPTION
**Pull request by henrinie-nc**
switches login to use HDS LoginProvider

Fix language selection with new hds-react (seems to require a heading, otherwise no options show *although not documented like that*)

### CODE REVIEW

File: src/auth/components/authDependentContent.tsx [hunk 1]
```diff
@@ -1,34 +1,35 @@
...
-  isLoadingApiToken,
+  isRenewingApiToken,
   user,
   hasApiToken,
   children,
-}: Props): JSX.Element | null =>
-  children(isLoadingUser || isLoadingApiToken, !!user && hasApiToken);
+}: Props): JSX.Element | null => {
+  // User exists and api token is not renewing
+  const isLoading = !user || isRenewingApiToken;
```
Comment (juho-kettunen-nc) [inline src/auth/components/authDependentContent.tsx]: Is the comment for this line correct? "User exists", but the logic says `!user`.

File: vite.config.ts [hunk 2]
```diff
@@ -49,7 +49,7 @@ export default defineConfig({
     }
   },
   envDir: './',
-  envPrefix: 'REACT_',
+  //envPrefix: 'REACT_',
```
Comment (juho-kettunen-nc) [inline vite.config.ts]: can this be removed entirely?

File: package.json [hunk 3]
```diff
@@ -31,7 +31,6 @@
     "leaflet-draw": "^1.0.4",
     "leaflet-geosearch": "^3.10.2",
     "leaflet-measure-path": "^1.5.0",
-    "oidc-client-ts": "^2.4.0",
```
Comment (juho-kettunen-nc) [inline package.json:34]: I assume it is coming as a nested dependency via hds?

**Review by juho-kettunen-nc — APPROVED** [review]

File: src/auth/components/authDependentContent.tsx (hunk 1 above)
Comment (henrinie-nc) [reply src/auth/components/authDependentContent.tsx]: Seems incorrect indeed

File: vite.config.ts (hunk 2 above)
Comment (henrinie-nc) [reply vite.config.ts]: Though I had this commited, indeed it is not needed

File: package.json (hunk 3 above)
Comment (henrinie-nc) [reply package.json:34]: That is correct

**Pull request merged**